### DISCUSSION TITLE
Creating mapsforge .Map file from non-OSM data

### DISCUSSION THREAD

**Discussion by Maikel86** [mapsforge/mapsforge#1358] [Q&A] [answered]
Hi, I have data from another source that I would like to bring to SMASH (geopaparazzi, Android) as a basemap. I am exploring the possibility of creating my own mapsforge map file with non-OSM data. Sorry if this is already an old topic. Since I'm new here, is there anyone able to advise about how difficult would it be?

**Comment by devemux86** [comment, marked as answer]
Can convert custom data (e.g. shp files) to osm as input to Osmosis.
Use custom tag-mapping with map-writer and custom render theme in app.

See the [map-writer documentation](https://github.com/mapsforge/mapsforge/blob/master/docs/Getting-Started-Map-Writer.md).
